frontend/src/store/middlewares/socket/chat-socket.middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { SocketEvent, SocketNamespace } from 'common/enums/enums';
+import {
+  AppDispatch,
+  ChatMessageGetAllItemResponseDto,
+} from 'common/types/types';
+import { io } from 'socket.io-client';
+import { chatsActions } from 'store/actions';
+
+type SocketMiddlewareParams = {
+  dispatch: AppDispatch;
+};
+
+const chatSocketInstance = io(SocketNamespace.CHAT);
+
+const chatSocket: Middleware =
+  ({ dispatch }: SocketMiddlewareParams) =>
+  (next) =>
+  (action): void => {
+    if (chatsActions.joinRoom.match(action)) {
+      chatSocketInstance.emit(SocketEvent.CHAT_JOIN_ROOM, action.payload);
+      chatSocketInstance.on(
+        SocketEvent.CHAT_ADD_MESSAGE,
+        (message: ChatMessageGetAllItemResponseDto): void => {
+          dispatch(chatsActions.addMessage(message));
```

Comment:
Move this outside if, backend should handle this logic with sending appropriate event only for needed room

Reply:
also let's keep all listeners at the bottom

Reply:
But in that case 1 user will start lots of listeners to the same event and when a user sends a message, another user gets the message the number of listeners times

Reply:
Yes, you are right, what if we define listeners two levels above  where we get dispatch?

Reply:
That seems to work :+1: